pulp_container/app/tasks/sync_stages.py

```diff
@@ -610,95 +664,37 @@ def _pad_unpadded_b64(self, unpadded_b64):
-            )
-
-            for dc in batch:
-                await self.put(dc)
-
-    def relate_config_blob(self, dc):
-        """
-        Relate a Blob to a Manifest as a config layer.
-
-        Args:
-            dc (pulpcore.plugin.stages.DeclarativeContent): dc for a Blob
-
-        Returns:
-            pulp_container.app.models.Manifest: An existing Manifest object with the updated
-                config layer
-
-        """
-        configured_dc = dc.extra_data.get("config_relation")
-        configured_dc.content.config_blob = dc.content
-        return configured_dc.content
-
-    def relate_blob(self, dc):
-        """
-        Relate a Blob to a Manifest.
-
-        Args:
-            dc (pulpcore.plugin.stages.DeclarativeContent): dc for a Blob
-
-        Returns:
-            pulp_container.app.models.BlobManifest: A new BlobManifest object with the added
-                reference to a Manifest object
-
-        """
-        related_dc = dc.extra_data.get("blob_relation")
-        return BlobManifest(manifest=related_dc.content, manifest_blob=dc.content)
-
-    def relate_manifest_to_list(self, dc):
-        """
-        Relate an ImageManifest to a ManifestList.
-
-        Args:
-            dc (pulpcore.plugin.stages.DeclarativeContent): dc for a ImageManifest
-
-        Returns:
-            pulp_container.app.models.ManifestListManifest: A new ManifestListManifest object
-                with the associated ImageManifest object and corresponding platform information
-
-        """
-        related_dc = dc.extra_data.get("relation")
-        platform = dc.extra_data.get("platform")
-        return ManifestListManifest(
-            manifest_list=dc.content,
-            image_manifest=related_dc.content,
-            architecture=platform["architecture"],
-            os=platform["os"],
-            features=platform.get("features"),
-            variant=platform.get("variant"),
-            os_version=platform.get("os.version"),
-            os_features=platform.get("os.features"),
-        )
+    def _post_save(self, batch):
+        blob_manifests = []
+        manifest_list_manifests = []
+        for dc in batch:
+            if "blob_dcs" in dc.extra_data:
+                blob_manifests.extend(
+                    (
+                        BlobManifest(manifest=dc.content, manifest_blob=blob_dc.content)
+                        for blob_dc in dc.extra_data["blob_dcs"]
+                    )
+                )
+            if "listed_manifests" in dc.extra_data:
+                for listed_manifest in dc.extra_data["listed_manifests"]:
+                    manifest_dc = listed_manifest["manifest_dc"]
+                    platform = listed_manifest["platform"]
+                    manifest_list_manifests.append(
+                        ManifestListManifest(
+                            manifest_list=manifest_dc.content,
+                            image_manifest=dc.content,
```

Comment:
Wow, i really think we are using these variables the wrong way around.

Reply:
How much effort will it take to rename those variables?

Reply:
It involves a data migration at least. I don't want to mix that up in another big PR like this.